config/routes.rb

```diff
@@ -1,3 +1,11 @@
 Rails.application.routes.draw do
-  resources :markets
+  root 'markets#home'
+  resources :markets do
+
+  end
+  resources :vendors do
+    resources :products do
```

Comment:
You can avoid using the `do` and `end` if you are not nesting any routes within a given resource